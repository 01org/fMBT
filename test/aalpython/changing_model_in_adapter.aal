aal "enabled_actions" {
    language: python {}

    # preview-hide-vars:
    # preview-depth: 5
    variables {
        x
    }
    initial_state {
        x = 0
    }
    action "iIncSix" {
        guard() { return x < 3 }
        body()  { x += 1 }
        adapter() {
            x += 5
        }
    }
    action "iDecOne" {
        guard() { return x == 6 }
        adapter() {
            x -= 1
            return action("iFive")
        }
    }
    action "iFive" {
        guard() { return x == 5 }
        adapter() {
            x = 4
            return action("iIncSix")
        }
    }
}
