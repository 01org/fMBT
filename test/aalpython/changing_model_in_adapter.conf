model     = "aal_remote(remote_pyaal -l enabled_actions.aal.log 'changing_model_in_adapter.aal')"
adapter   = "aal"
heuristic = "lookahead(4)"
coverage  = "perm(2)"
pass      = "steps(20)"
on_pass   = "exit(0)"
on_fail   = "exit(1)"
on_inconc = "exit(2)"