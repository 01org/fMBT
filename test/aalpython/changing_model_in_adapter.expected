iIncSix -> iIncSix
 
iDecOne -> iFive
 
iFive -> iIncSix
fail unexpected response to input "iFive": "iIncSix".
